lms/resources/_lti_launch.py

```diff
@@ -154,6 +157,29 @@ def h_userid(self):
         """
         return f"acct:{self.h_username}@{self._authority}"
 
+    @property
+    def js_config(self):
+        """
+        Return the configuration for the app's JavaScript code.
+
+        This is a mutable config dict. It can be accessed, for example by
+        views, as ``request.context.js_config``, and they can mutate it or add
+        their own view-specific config settings. The modified config object
+        will then be passed to the JavaScript code in the response page.
+
+        :rtype: dict
+        """
+        if self._js_config is None:
+            # Initialize self._js_config for the first time.
+            self._js_config = {"urls": {}}
```

Comment:
What is "urls" going to be used for?

Reply:
This was intended to hold all the URLs that the JS needs to do its job. Instead of passing parameters like the course_id, file_id and authentication param, the server would put the file list, via url, and popup window URLs with the params already in them. So JS doesn't need to construct URLs. So far though we've forgotten to use it! But I intend to start doing so

Given that our proxy API already has to accept the JWT auth param as a query param (because of the popup window) we _could_ also put it as a query param rather than a header in the XHR API requests as well, meaning the server could render the URLs with the param already on them, then the JS wouldn't have to worry about sending auth headers either